docs/traffic-control.md

```diff
@@ -0,0 +1,277 @@
+set to the port to which the traffic will be mirrored. For the `Redirect`
+action, both `targetPort` and `returnPort` need to be specified, the latter of
+which represents the port from which the traffic will be sent back to OVS and be
+forwarded to its original destination.
+
+### TargetPort
+
+The `targetPort` field specifies the port to which the traffic should be
+redirected or mirrored. There are five kinds of ports that can be used to
+receive mirrored traffic:
+
+**ovsInternal**: This specifies an OVS internal port on all Nodes. A Pod's
+traffic will be redirected or mirrored to the OVS internal port on the same Node
+that hosts the Pod. The port doesn't need to exist in advance, Antrea will
+create the port if it doesn't exist. To use an OVS internal port, the `name` of
+the port must be provided:
+
+```yaml
+ovsInternal:
+  name: tap0
+```
+
+**device**: This specifies a network device on all Nodes. A Pod's traffic will
+be redirected or mirrored to the network device on the same Node that hosts the
+Pod. The network device must exist on all Nodes and Antrea will attach it to the
+OVS bridge if not already attached. To use a network device, the `name` of the
+device must be provided:
+
+```yaml
+device:
+  name: eno2
+```
+
+**geneve**: This specifies a remote destination for a GENEVE tunnel. All
+selected Pods' traffic will be redirected or mirrored to the destination via
+a GENEVE tunnel. The `remoteIP` field must be provided to specify the IP address
+of the destination. Optionally, the `destinationPort` field could be used to
+specify the UDP destination port of the tunnel, or 6081 will be used by default.
+If Virtual Network Identifier (VNI) is desired, the `vni` field can be specified
+to an integer in the range 0-16,777,215:
+
+```yaml
+geneve:
+  remoteIP: 10.0.10.2
+  destinationPort: 6081
+  vni: 1
+```
+
+**vxlan**: This specifies a remote destination for a VXLAN tunnel. All
+selected Pods' traffic will be redirected or mirrored to the destination via
+a VXLAN tunnel. The `remoteIP` field must be provided to specify the IP address
+of the destination. Optionally, the `destinationPort` field could be used to
+specify the UDP destination port of the tunnel, or 4789 will be used by default.
+If Virtual Network Identifier (VNI) is desired, the `vni` field can be specified
+to an integer in the range 0-16,777,215:
+
+```yaml
+vxlan:
+  remoteIP: 10.0.10.2
+  destinationPort: 4789
+  vni: 1
+```
+
+**gre**: This specifies a remote destination for a GRE tunnel. All selected
+Pods' traffic will be redirected or mirrored to the destination via a GRE
+tunnel. The `remoteIP` field must be provided to specify the IP address of the
+destination. If GRE key is desired, the `key` field can be specified to an
+integer in the range 0-4,294,967,295:
+
+```yaml
+gre:
+  remoteIP: 10.0.10.2
+  key: 1
+```
+
+**erspan**: This specifies a remote destination for an ERSPAN tunnel. All
```

Comment:
Should we mention the requirement of ERSPAN?

It can be used with 4.19-rc6+ kernel and iproute2-ss180813?

Reply:
Page 19 in http://vger.kernel.org/lpc_net2018_talks/erspan-linux-presentation.pdf

Reply:
It's an "Linux Native Mode tunnel example" with 4.19-rc6+ kernel and iproute2-ss180813, not saying the supported is added in that version and iproute2 is required for OVS ERSPAN.

Reply:
Got it, for OVS ERSPAN, iproute2 is not related, but according to this http://vger.kernel.org/lpc_net2018_talks/erspan-linux.pdf, `ERSPAN was added to Linux kernel in 4.14 for IPv4 and 4.16 for IPv6.`, we should mention this?

Reply:
but I remember you said 4.15 didn't work in your testbed? Unless we have accurate information, I wouldn't mention its version requirement given it's in alpha phase, to avoid misleading users. We can add the information anytime once we have verified it.

Reply:
Yes, it didn't work in 4.15 in my testbed actually. Maybe we can add the information in the future.